PartDesign: Enabling "Allow multiple solids in Part Design Body by default" breaks testUsagePocketInternalGearProfile

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Latest main.

Steps to reproduce:

1. Start with fresh empty config
2. Run FreeCAD --run-test 0  - to run all tests  (or FreeCADCmd)
3. All tests should succeed
4. Open FreeCAD
5. In Edit->Preferences-->PartDesign->General->Experimental enable Allow multiple solids in Part Design Body by default
6. Close app and rerun tests

**Expected outcome:**

All tests succeed

**Actual outcome:**

```
=====================================================================
FAIL: testUsagePocketInternalGearProfile (PartDesignTests.TestInvoluteGear.TestInvoluteGear)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/slow1/src/FreeCAD/main/Mod/PartDesign/PartDesignTests/TestInvoluteGear.py", line 268, in testUsagePocketInternalGearProfile
    self.assertSolid(pocket.Shape)
  File "/mnt/slow1/src/FreeCAD/main/Mod/PartDesign/PartDesignTests/TestInvoluteGear.py", line 327, in assertSolid
    self.assertEqual(shape.ShapeType, 'Solid', msg=msg)
AssertionError: 'Compound' != 'Solid'
- Compound
+ Solid
```


### Full version info

```shell
OS: Ubuntu 20.04.6 LTS (XFCE/xfce/xcb)
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.38991 (Git)
Build type: Debug
Branch: main
Hash: ef2156d0f6d116b1642a5531a2d44fc6b7d17042
Python 3.8.10, Qt 5.12.8, Coin 4.0.0, Vtk 7.1.1, OCC 7.6.3
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
```


### Subproject(s) affected?

PartDesign

### Anything else?

this is one of multiple issues caused by user config - probably a harmless one.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Theres another one:  PartDesignTests.TestMultiTransform.TestMultiTransform.testMultiTransformDressup breaks wiith a geometry flip:

```
======================================================================
FAIL: testMultiTransformDressup (PartDesignTests.TestMultiTransform.TestMultiTransform)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/slow1/src/FreeCAD/main/Mod/PartDesign/PartDesignTests/TestMultiTransform.py", line 123, in testMultiTransformDressup
    self.assertAlmostEqual(Body.Shape.Volume, 990)
AssertionError: -990.0 != 990 within 7 places (1980.0 difference)
```

If you do BOTH
1, In _Edit->Preferences-->PartDesign->General->Experimental_ ENABLE **Allow multiple solids in Part Design Body by default**
2.  In _Edit->Preferences-->PartDesign->General->Model Settings_   DISABLE **Automatically refine model after sketch-based operation**

I add it here since its closely related

--- a/src/Mod/PartDesign/PartDesignTests/TestInvoluteGear.py
+++ b/src/Mod/PartDesign/PartDesignTests/TestInvoluteGear.py
@@ -254,6 +254,7 @@ class TestInvoluteGear(unittest.TestCase):
         profile.HighPrecision = False
         profile.NumberOfTeeth = 8
         body = self.Doc.addObject('PartDesign::Body','GearBody')
+        body.AllowCompound = False
         body.addObject(profile)
         cylinder = body.newObject('PartDesign::AdditiveCylinder','GearCylinder')
         default_dedendum = 1.25

--- a/src/Mod/PartDesign/PartDesignTests/TestMultiTransform.py
+++ b/src/Mod/PartDesign/PartDesignTests/TestMultiTransform.py
@@ -70,6 +70,7 @@ class TestMultiTransform(unittest.TestCase):
         # Arrange
         Doc = self.Doc
         Body = Doc.addObject('PartDesign::Body','Body')
+        Body.AllowCompound = False
         # Make first offset cube Pad
         PadSketch = Doc.addObject('Sketcher::SketchObject', 'SketchPad')
         Body.addObject(PadSketch)

